src/core/variable.jl

```diff
@@ -150,18 +150,10 @@ function variable_mc_branch_flow_reactive(pm::_PMs.AbstractPowerModel; nw::Int=p
     )
 
     if bounded
-        for c in cnds
-            flow_lb, flow_ub = _PMs.ref_calc_branch_flow_bounds(_PMs.ref(pm, nw, :branch), _PMs.ref(pm, nw, :bus), c)
-
-            for arc in _PMs.ref(pm, nw, :arcs)
-                l,i,j = arc
-                if !isinf(flow_lb[l])
-                    JuMP.set_lower_bound(q[arc][c], flow_lb[l])
-                end
-                if !isinf(flow_ub[l])
-                    JuMP.set_upper_bound(q[arc][c], flow_ub[l])
-                end
-            end
+        for (l,i,j) in _PMs.ref(pm, nw, :arcs)
+            smax = _calc_branch_power_max(_PMs.ref(pm, nw, :branch, l), _PMs.ref(pm, nw, :bus, i))
+            JuMP.set_upper_bound.(q[(l,i,j)],  smax)
+            JuMP.set_lower_bound.(q[(l,i,j)], -smax)
```

Comment:
should we instead be using the internal `set_lower_bound` and `set_upper_bound` here?